### PR TITLE
Improvement

### PR DESCRIPTION
**Pull request by diocas**
First question: why do we need a `swan_user` label if JH already provides something similar? 

If we need it, then we need to escape it like kubespawner does (we had issues in a sciencebox deployment).

This commit also removes some leftovers that are not used.

### CODE REVIEW

File: swan/files/swan_config.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import logging, os, subprocess
+import logging
+import escapism
```
Comment (etejedor) [inline swan/files/swan_config.py]: To be purist, this would need to be in the other commit.

File: swan/files/swan_config.py [hunk 2]
```diff
@@ -24,10 +24,14 @@ def __init__(self, spawner, pod):
...
 
+        safe_username = escapism.escape(
+            self.spawner.user.name, safe = self.spawner.safe_chars, escape_char = '-'
+        ).lower()
+
         # pod labels
         pod_labels = dict(
             lcg_release = self.spawner.user_options[self.spawner.lcg_rel_field].split('/')[0],
-            swan_user = self.spawner.user.name
+            swan_user = safe_username # FIXME why do we need a swan_user if we have username already?
```
Comment (etejedor) [inline swan/files/swan_config.py]: I guess it is not strictly needed since the pod name already contains the user name, it is just a label that is added to the user pod to add some extra metadata. Perhaps if a user had multiple servers it would be useful to filter only those of the users. I guess it doesn't hurt to keep it.

But why do you need to escape the user name before using it in a label?

Comment (diocas) [reply swan/files/swan_config.py]: Because it can contain invalid chars. I.e, external accounts in our SSO have the username in the format "some_id@service_name" (and @ is invalid).
It's good practice.

But for the main thing: we already have the username as label (`hub.jupyter.org/username`), that's why I asked.

File: swan/files/swan_config.py (hunk 1 above)
Comment (diocas) [reply swan/files/swan_config.py]: Yes, it was a mistake

File: swan/files/swan_config.py (hunk 2 above)
Comment (etejedor) [reply swan/files/swan_config.py]: Then I think we can get rid of that label, I don't see where it is used -- but it would be good if you could try first, to see if it breaks something evident at least.

Even if we remove this instance, there is also [this](https://github.com/swan-cern/swan-charts/blob/master/swan-cern/files/swan_config_cern.py#L52) where an equivalent treatment should be applied.

File: swan-cern/files/swan_config_cern.py [hunk 3]
```diff
@@ -46,38 +41,44 @@ async def get_swan_user_pod(self):
...
 
-        # ITHADOOP-819 - Ports need to be opened using service creation, and later assigning allocated service nodeport to a pod 
+        # ITHADOOP-819 - Ports need to be opened using service creation, and later assigning allocated service nodeport to a pod
         # Create V1Secret with eos token
         secret_data = V1Secret()
 
         secret_meta = V1ObjectMeta()
         secret_meta.name = eos_secret_name
         secret_meta.namespace = swan_container_namespace
-        secret_meta.labels = {
```
Comment (etejedor) [inline swan-cern/files/swan_config_cern.py]: I didn't mean to remove this label from here (this is a secret, this is not the user pod). I meant that it should be escaped.

File: swan-cern/files/swan_config_cern.py [hunk 4]
```diff
@@ -46,38 +41,44 @@ async def get_swan_user_pod(self):
 
     async def _init_eos_secret(self):
         username = self.spawner.user.name
-        user_uid = self.spawner.user_uid
-        eos_secret_name ='eos-tokens-%s' % username
+        eos_secret_name = "eos-tokens-%s" % username
```
Comment (etejedor) [inline swan-cern/files/swan_config_cern.py]: I'd rather not have all these formatting changes, this is just going to create some noise and anyway the rest of files are not formatted.

File: swan/files/swan_config.py [hunk 5]
```diff
@@ -26,8 +25,7 @@ def get_swan_user_pod(self):
 
         # pod labels
         pod_labels = dict(
-            lcg_release = self.spawner.user_options[self.spawner.lcg_rel_field].split('/')[0],
-            swan_user = self.spawner.user.name
```
Comment (etejedor) [inline swan/files/swan_config.py:30]: Can I ask one last thing? Separate this change into another commit, so you can explain that we remove this label because there is another label (you should say which one) that contains the user name. And that we don't seem to be using the current label for anything.

**Review by etejedor — APPROVED** [review]
Many thanks!
